dynamo/tools/utils.py

```diff
@@ -1590,25 +1597,96 @@ def set_param_kinetic(
+    params_df.loc[valid_ind, kin_param_pre + "half_life"] = np.log(2) / gamma
+    params_df.loc[valid_ind, kin_param_pre + "cost"] = cost
+    params_df.loc[valid_ind, kin_param_pre + "logLL"] = logLL
     # add extra parameters (u0, uu0, etc.)
     extra_params.columns = [kin_param_pre + i for i in extra_params.columns]
     extra_params = extra_params.set_index(adata.var.index[valid_ind])
-    var = pd.concat((adata.var, extra_params), axis=1, sort=False)
-    adata.var = var
+    var = pd.concat((params_df, extra_params), axis=1, sort=False)
+    adata.varm[kin_param_pre + "vel_params"] = var.to_numpy()
+    adata.uns[kin_param_pre + "vel_params_names"] = list(var.columns)
 
     return adata
 
 
+def get_vel_params(
+    adata: AnnData,
+    params: Optional[Union[List, str]] = None,
+    genes: Optional[List] = None,
+    kin_param_pre: str = "",
+    skip_cell_wise: bool = False,
+) -> Union[Tuple, pd.DataFrame, List]:
+    """Get the velocity parameters based on input names.
+
+    Args:
+        adata: the anndata object which contains the parameters.
+        params: the names of parameters to query. If set to None, the entire velocity parameters DataFrame from `.varm`
+            will be returned.
+        kin_param_pre: the prefix used in dynamics when estimating the parameters.
+        skip_cell_wise: whether to skip the detected cell wise parameters. If set to True, the mean will be returned
+            instead of cell wise parameters.
+
+    Returns:
+        All velocity parameters with the same order of query `params`.
+    """
+    if type(params) is str:
+        params = [params]
+
+    if kin_param_pre + "vel_params" not in adata.varm.keys():
+        raise KeyError("No velocity parameters found.")
+
+    array_data = adata.varm[kin_param_pre + "vel_params"]
+    df_columns = adata.uns[kin_param_pre + "vel_params_names"]
+    df = pd.DataFrame(array_data, index=adata.var_names, columns=df_columns)
+    target_params = []
+
+    if genes is None:
+        genes = df.index
+
+    if params is None:
+        return df.loc[genes]
+
+    for param in params:
+        if param == "alpha":
+            if not skip_cell_wise:
+                if "cell_wise_alpha" in adata.layers.keys():
+                    target_params.append(adata[:, genes].layers["cell_wise_alpha"])
+                elif "alpha" in adata.varm.keys():
+                    target_params.append(adata[:, genes].varm[kin_param_pre + "alpha"])
+                else:
+                    target_params.append(df.loc[genes, kin_param_pre + "alpha"].values)
+                continue
+        target_params.append(df.loc[genes, kin_param_pre + param].values)
+
+    if len(target_params) > 1:
+        return tuple(target_params)
+    else:
+        return target_params[0]
+
+
+def update_vel_params(adata: AnnData, params_df: pd.DataFrame, kin_param_pre: str = "") -> None:
+    """Update the velocity parameters.
+
+    Args:
+        adata: the AnnData obejct to update.
```

Comment:
please make sure your documentation to be precise instead of lazy! 
adata: the AnnData obejct whose kinetic parameters related RNA velocity calculation will be update.        
params_df: the dataframe of kinetic parameters related RNA velocity calculation.
kin_param_pre: the prefix of the key to locate the corresponding parameters related to RNA velocity calculation.